site/src/job_queue.rs

```diff
@@ -51,8 +186,412 @@ pub async fn cron_main(site_ctxt: Arc<RwLock<Option<Arc<SiteCtxt>>>>, seconds: u
             guard.as_ref().cloned()
         } {
-            cron_enqueue_jobs(&ctxt_clone).await;
-            log::info!("Cron job executed at: {:?}", std::time::SystemTime::now());
+            match cron_enqueue_jobs(&ctxt_clone).await {
+                Ok(_) => log::info!("Cron job executed at: {:?}", std::time::SystemTime::now()),
+                Err(e) => log::error!("Cron job failed to execute {}", e),
+            }
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use chrono::{Datelike, Duration, TimeZone, Utc};
+
+    use database::tests::run_postgres_test;
+
+    fn days_ago(day_str: &str) -> chrono::DateTime<Utc> {
+        // Walk backwards until the first non-digit, then slice
+        let days = day_str
+            .strip_prefix("days")
+            .unwrap()
+            .parse::<i64>()
+            .unwrap();
+
+        let timestamp = Utc::now() - Duration::days(days);
+        // zero out the seconds
+        Utc.with_ymd_and_hms(
+            timestamp.year(),
+            timestamp.month(),
+            timestamp.day(),
+            0,
+            0,
+            0,
+        )
+        .unwrap()
+    }
+
+    fn create_master(sha: &str, parent: &str, pr: u32, age_days: &str) -> BenchmarkRequest {
+        BenchmarkRequest::create_master(
+            sha,
+            parent,
+            pr,
+            days_ago(age_days),
+            BenchmarkRequestStatus::ArtifactsReady,
+            "",
+            "",
+        )
+    }
+
+    fn create_try(sha: &str, parent: &str, pr: u32, age_days: &str) -> BenchmarkRequest {
+        BenchmarkRequest::create_try(
+            sha,
+            parent,
+            pr,
+            days_ago(age_days),
+            BenchmarkRequestStatus::ArtifactsReady,
+            "",
+            "",
+        )
+    }
+
+    fn create_release(tag: &str, age_days: &str) -> BenchmarkRequest {
+        BenchmarkRequest::create_release(
+            tag,
+            days_ago(age_days),
+            BenchmarkRequestStatus::ArtifactsReady,
+            "",
+            "",
+        )
+    }
+
+    async fn db_insert_requests(
+        conn: &dyn database::pool::Connection,
+        requests: &[BenchmarkRequest],
+    ) {
+        for request in requests {
+            conn.insert_benchmark_request(&request).await;
+        }
+    }
+
+    /// Get an `InProgress` item out of the `benchmark_requests` table. In
+    /// practice this is the job that has been enqueued.
+    async fn get_in_progress(conn: &dyn database::pool::Connection) -> Option<BenchmarkRequest> {
+        conn.get_benchmark_requests_by_status(&[BenchmarkRequestStatus::InProgress], None)
+            .await
+            .unwrap()
+            .first()
+            .cloned()
+    }
+
+    fn queue_order_matches(queue: &[BenchmarkRequest], expected: &[&str]) -> bool {
+        let queue_shas: Vec<&str> = queue.iter().map(|req| req.tag()).collect();
+        if queue_shas.len() != expected.len() {
```

Comment:
Maybe just `assert_eq!(expected, &queue_shas)`? The return value probably also isn't needed.